crates/sui-package-management/src/lib.rs

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{anyhow, bail};
+use std::path::PathBuf;
+
+use move_package::lock_file::{self, LockFile};
+use sui_json_rpc_types::{get_new_package_obj_from_response, SuiTransactionBlockResponse};
+use sui_sdk::wallet_context::WalletContext;
+
+pub enum LockCommand {
+    Publish,
+    Upgrade,
+}
+
+pub async fn update_lock_file(
+    context: &WalletContext,
+    command: LockCommand,
+    install_dir: Option<PathBuf>,
+    lock_file: Option<PathBuf>,
+    response: &SuiTransactionBlockResponse,
+) -> Result<(), anyhow::Error> {
```

Comment:
Main function called from the CLI commands `publish`/`upgrade` to write to the lock.

Reply:
Let's add a doc comment to it to explain what it does.